api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -676,6 +676,12 @@ message Cluster {
     // If this is unset then [UNKNOWN, HEALTHY, DEGRADED] will be applied by default. If this is
     // set with an empty set of statuses then host overrides will be ignored by the load balancing.
     core.v3.HealthStatusSet override_host_status = 8;
+
+    // This controls what policy is used to extract the override host from the downstream context
+    // like request headers, cookies, dynamic metadata, etc.
+    // If the the override host is evaluated as valid host based on the ``override_host_status``,
+    // the host will be selected directly and the load balancing will be skipped.
+    core.v3.TypedExtensionConfig override_host_policy = 9;
```

Comment:
How does this interact with the existing internal API? By my read below if this is specified it will override any internal use in the code. Is this going to be confusing? Should the old uses be moved over to this?

Reply:
> How does this interact with the existing internal API? By my read below if this is specified it will override any internal use in the code. Is this going to be confusing? Should the old uses be moved over to this?

Yeah, I will gradually move the old uses (mostly stateful session filter) to this new policy and deprecate the old extension. By this way, we can gradually deprecate old uses.